packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -244,7 +244,15 @@ bool _writeFlutterPluginsList(
   final String? oldPluginsFileStringContent = _readFileContent(pluginsFile);
   bool pluginsChanged = true;
   if (oldPluginsFileStringContent != null) {
-    pluginsChanged = oldPluginsFileStringContent.contains(pluginsMap.toString());
+    Object? decodedJson;
+    try {
+      decodedJson = jsonDecode(oldPluginsFileStringContent);
+      if (decodedJson is Map<String, Object?>) {
+        final String jsonOfNewPluginsMap = jsonEncode(pluginsMap);
```

Comment:
should we like ... sort it?

Reply:
In theory it is sorted here:

https://github.com/flutter/flutter/blob/9ffdd3a0ecfa94b94a2fa856fc94e4fcbb2f50c6/packages/flutter_tools/lib/src/flutter_plugins.dart#L1107-L1108

Reply:
good enough for me